### DISCUSSION TITLE
Type-safety for process.env

### DISCUSSION THREAD

**Discussion by phibersoft** [vercel/next.js#65588] [Ideas]
### Goals

1. Avoiding usage of process.env? or process.env.KEY!
2. Initially making type-safe process.env

### Background

In every project, i have to create additional type file and update that file whenever i add something to environment files. We can change this for typescript developers

### Proposal

As a Node + Typescript user, I'm solving like this:
1. Create a `global.d.ts` file:
```typescript
export declare global {
    namespace NodeJS {
        interface ProcessEnv {
            CUSTOM_KEY: string;
        }
    }
}
```
2. Add this file path to `tsconfig.json > include`
3. Write same keys to this file whenever i change something in .env.local

NextJS already watching env files for changes, so maybe it can create a file on background and update it automatically?

**Comment by Netail** [comment]
Will be a feature in Next 15; https://github.com/vercel/next.js/pull/67157 :)

**Reply by phibersoft** [reply]
Awesome! Thanks for notifying 🙂

**Reply by HasanMothaffar** [reply]
Is this available in canary now or something? There are no official docs on type safety for env vars
